pkg/download/mode/mode.go

```diff
@@ -18,13 +19,17 @@ type Mode string
 
 // DownloadMode constants. For more information see config.dev.toml
 const (
-	Sync          Mode = "sync"
-	Async         Mode = "async"
-	Redirect      Mode = "redirect"
-	AsyncRedirect Mode = "async_redirect"
-	None          Mode = "none"
+	Sync           Mode = "sync"
+	Async          Mode = "async"
+	Redirect       Mode = "redirect"
+	AsyncRedirect  Mode = "async_redirect"
+	None           Mode = "none"
+	invalidModeErr      = "unrecognized download mode: %s"
 )
 
+// ErrNoDownloadMode is returned if the specified mode is empty string
+var ErrNoDownloadMode = e.New("download mode is not set")
```

Comment:
Given our `errors` package, this only needs to be a string, our errors package will actually call `errors.New` for you here: https://github.com/gomods/athens/blob/master/pkg/errors/errors.go#L96 so there's no need to import the errors package here. You can also put this under `invalidModeErr`